PIL/MpoImagePlugin.py

```diff
@@ -71,14 +71,13 @@ def is_animated(self):
         return self.__framecount > 1
 
     def seek(self, frame):
-        if frame < 0 or frame >= self.__framecount:
-            raise EOFError("no more images in MPO file")
-        else:
-            self.fp = self.__fp
-            self.offset = self.__mpoffsets[frame]
-            self.tile = [
-                ("jpeg", (0, 0) + self.size, self.offset, (self.mode, ""))
-            ]
+        if not self._seek_check(frame):
+            return
```

Comment:
The seek check here never fails in the testing. This one is checking __framecount, the core is checking n_frames